packages/perspective-viewer-d3fc/src/js/tooltip/tooltip.js

```diff
@@ -52,14 +52,36 @@ function getTooltipDiv(container) {
 function showTooltip(containerNode, barNode, tooltipDiv) {
     const containerRect = containerNode.getBoundingClientRect();
     const barRect = barNode.getBoundingClientRect();
+
     const left = barRect.left + barRect.width / 2 - containerRect.left;
     const top = barRect.top - containerRect.top;
+
     tooltipDiv
         .style("left", `${left}px`)
         .style("top", `${top}px`)
         .transition()
         .duration(200)
         .style("opacity", 0.9);
+
+    shiftIfOverflowingChartArea(tooltipDiv, containerRect, left, top);
+}
+
+function shiftIfOverflowingChartArea(tooltipDiv, containerRect, left, top) {
+    const tooltipDivRect = tooltipDiv._groups[0][0].getBoundingClientRect();
```

Comment:
The `node()` function returns the first node in the group. i.e.

````javascript
const tooltipDivRect = tooltipDiv.node().getBoundingClientRect();
````

Reply:
Useful to know